src/main/java/software/amazon/awssdk/crt/http/HttpException.java

```diff
@@ -6,13 +6,14 @@
 package software.amazon.awssdk.crt.http;
 
 import software.amazon.awssdk.crt.CRT;
+import software.amazon.awssdk.crt.CrtErrorInfo;
 
 /**
  * This exception will be thrown by any exceptional cases encountered within the
  * JNI bindings to the AWS Common Runtime
  */
 public class HttpException extends RuntimeException {
```

Comment:
My real dream for aws-crt-java would be to code-gen actual exception **classes** (not just enums) so Java programmers could do the idiomatic thing and:
```
catch (TlsNegotiationFailure e) {
    ...special...
}
catch (Exception e) {
    ...everything else...
}
```

instead of:
```
catch (HttpException e) {
    switch(e.getErrorInfo()) {
        case CrtErrorInfo.TLSNegotiationFailure:
            ...special...
            break;
        default:
            ...everything else (block 1)...
            break;
    }
}
catch (Exception e) {
    ...everything else (block 2)...
}
```

But that would technically be a breaking change, since we'd be changing the exact exception classes that get raised 😓

Would it be worth breaking those eggs?

Reply:
deferring this convo to a later time. For now I can get by with the errir code being made public on the exception class